File: src/dashboard.client/src/app/shared/models/results/triangle-result.ts
import { VertexDisplacements } from "./vertex-displacements";
import { VertexStresses } from "./vertex-stresses";


export interface TriangleResult {
  vertex1Displacements: VertexDisplacements;
  vertex2Displacements: VertexDisplacements;
  vertex3Displacements: VertexDisplacements;
  vertex1Stresses: VertexStresses;
  vertex2Stresses: VertexStresses;
  vertex3Stresses: VertexStresses;
}

File: src/Contracts/TriangleResult.cs
namespace Contracts
{
    public record struct TriangleResult
    {
        public NodeDisplacementResult Vertex1Displacements { get; init; }
        public NodeDisplacementResult Vertex2Displacements { get; init; }
        public NodeDisplacementResult Vertex3Displacements { get; init; }
        public NodeStressResult Vertex1Stresses { get; init; }
        public NodeStressResult Vertex2Stresses { get; init; }
        public NodeStressResult Vertex3Stresses { get; init; }
    }
}
